### PR TITLE
Add tests to fail participants during repair (CASSANDRA-12901)

### PR DESCRIPTION
**Pull request by pauloricardomg**

### CODE REVIEW

File: repair_tests/repair_test.py [hunk 1]
```diff
@@ -1054,6 +1054,94 @@ def test_dead_coordinator(self):
...
+        t = Thread(target=node1_repair)
+        t.start()
+
+        debug("Will kill {} in middle of {}".format(node_to_kill.name, phase))
+        msg_to_wait = 'streaming plan for Repair'
+        if phase == 'anticompaction':
+            msg_to_wait = 'Got anticompaction request'
+        elif phase == 'validation':
+            msg_to_wait = 'Validating'
+        node_to_kill.watch_log_for(msg_to_wait, filename='debug.log')
```
Comment (ptnapoleon) [inline repair_tests/repair_test.py]: This won't work on 2.1, we don't have `debug.log` there

Comment (pauloricardomg) [inline repair_tests/repair_test.py]: Thanks, this test requires byteman and is for 2.2+ anyway. I will add a `since(2.2)` annotation.

**Comment by pauloricardomg** [comment]
Added `@since('2.2')` annotation to new tests.

**Comment by pauloricardomg** [comment]
CASSANDRA-12901 merged

**Pull request merged**